<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsArrowSymbolLayerWidgetBase</class>
 <widget class="QWidget" name="QgsArrowSymbolLayerWidgetBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>721</width>
    <height>364</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="0,1,0">
   <item row="1" column="2">
    <widget class="QgsPropertyOverrideButton" name="mArrowTypeDDBtn">
     <property name="text">
      <string>…</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QComboBox" name="mHeadTypeCombo">
     <item>
      <property name="text">
       <string>Single</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Single, reversed</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Double</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QCheckBox" name="mCurvedArrowChck">
     <property name="text">
      <string>Curved arrows</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="tristate">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="9" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="5" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QgsDoubleSpinBox" name="mHeadThicknessSpin">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="maximum">
        <double>999999999.999999046325684</double>
       </property>
       <property name="singleStep">
        <double>0.200000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QgsUnitSelectionWidget" name="mHeadThicknessUnitWidget" native="true">
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="5" column="2">
    <widget class="QgsPropertyOverrideButton" name="mHeadHeightDDBtn">
     <property name="text">
      <string>…</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>Offset</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>Arrow type</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>Arrow width</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="4" column="2">
    <widget class="QgsPropertyOverrideButton" name="mHeadWidthDDBtn">
     <property name="text">
      <string>…</string>
     </property>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="QgsPropertyOverrideButton" name="mArrowWidthDDBtn">
     <property name="text">
      <string>…</string>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QgsDoubleSpinBox" name="mOffsetSpin">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="minimum">
        <double>-99999999.989999994635582</double>
       </property>
       <property name="maximum">
        <double>99999999.989999994635582</double>
       </property>
       <property name="singleStep">
        <double>0.200000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QgsUnitSelectionWidget" name="mOffsetUnitWidget" native="true">
       <property name="focusPolicy">
        <enum>Qt::StrongFocus</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="6" column="2">
    <widget class="QgsPropertyOverrideButton" name="mOffsetDDBtn">
     <property name="text">
      <string>…</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QgsDoubleSpinBox" name="mArrowWidthSpin">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="maximum">
        <double>999999999.999999046325684</double>
       </property>
       <property name="singleStep">
        <double>0.200000000000000</double>
       </property>
       <property name="showClearButton" stdset="0">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QgsUnitSelectionWidget" name="mArrowWidthUnitWidget" native="true">
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="2">
    <widget class="QgsPropertyOverrideButton" name="mHeadTypeDDBtn">
     <property name="text">
      <string>…</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Head thickness</string>
     </property>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="QgsPropertyOverrideButton" name="mArrowStartWidthDDBtn">
     <property name="text">
      <string>…</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label_9">
     <property name="text">
      <string>Head length</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="4" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QgsDoubleSpinBox" name="mHeadLengthSpin">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="maximum">
        <double>999999999.999999046325684</double>
       </property>
       <property name="singleStep">
        <double>0.200000000000000</double>
       </property>
       <property name="showClearButton" stdset="0">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QgsUnitSelectionWidget" name="mHeadLengthUnitWidget" native="true">
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label_10">
     <property name="text">
      <string>Head type</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QComboBox" name="mArrowTypeCombo">
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Plain: the arrow will be displayed entirely&lt;/p&gt;&lt;p&gt;Left/Exterior half: only the half of the head that is on the left of the arrow for straight arrows, or the one toward the exterior for curved arrows will be displayed&lt;/p&gt;&lt;p&gt;Right/Interior half: only the half of the head that is on the right of the arrow for straight arrows, or the one toward the interior for curved arrows will be displayed&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <item>
      <property name="text">
       <string>Plain</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Left/Exterior half</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Right/Interior half</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Arrow width at start</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <widget class="QgsDoubleSpinBox" name="mArrowStartWidthSpin">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="maximum">
        <double>999999999.999999046325684</double>
       </property>
       <property name="singleStep">
        <double>0.200000000000000</double>
       </property>
       <property name="showClearButton" stdset="0">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QgsUnitSelectionWidget" name="mArrowStartWidthUnitWidget" native="true">
       <property name="focusPolicy">
        <enum>Qt::TabFocus</enum>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="8" column="1">
    <widget class="QCheckBox" name="mRepeatArrowChck">
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;If checked, one arrow will be rendered for each consecutive points (each 2 points for a straight arrow or 3 points for a curved arrow).&lt;/p&gt;&lt;p&gt;If unchecked, the arrow will be defined by extermum points of the line (the middle point will be used as a control point for a curved arrow)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="text">
      <string>Repeat arrow on each segment</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <property name="tristate">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>qgsdoublespinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsPropertyOverrideButton</class>
   <extends>QToolButton</extends>
   <header>qgspropertyoverridebutton.h</header>
  </customwidget>
  <customwidget>
   <class>QgsUnitSelectionWidget</class>
   <extends>QWidget</extends>
   <header>qgsunitselectionwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>mHeadTypeCombo</tabstop>
  <tabstop>mHeadTypeDDBtn</tabstop>
  <tabstop>mArrowTypeCombo</tabstop>
  <tabstop>mArrowTypeDDBtn</tabstop>
  <tabstop>mArrowWidthSpin</tabstop>
  <tabstop>mArrowWidthUnitWidget</tabstop>
  <tabstop>mArrowWidthDDBtn</tabstop>
  <tabstop>mArrowStartWidthSpin</tabstop>
  <tabstop>mArrowStartWidthUnitWidget</tabstop>
  <tabstop>mArrowStartWidthDDBtn</tabstop>
  <tabstop>mHeadLengthSpin</tabstop>
  <tabstop>mHeadLengthUnitWidget</tabstop>
  <tabstop>mHeadWidthDDBtn</tabstop>
  <tabstop>mHeadThicknessSpin</tabstop>
  <tabstop>mHeadThicknessUnitWidget</tabstop>
  <tabstop>mHeadHeightDDBtn</tabstop>
  <tabstop>mOffsetSpin</tabstop>
  <tabstop>mOffsetUnitWidget</tabstop>
  <tabstop>mOffsetDDBtn</tabstop>
  <tabstop>mCurvedArrowChck</tabstop>
  <tabstop>mRepeatArrowChck</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
